torch/_ops.py

```diff
@@ -336,6 +349,11 @@ def handler(*args, **kwargs):
             return handler
 
         key = resolve_key(self, key)
+
+        if key is torch._C.DispatchKey.Autograd:
```

Comment:
Do we want to special case this for Autograd?

Reply:
I'm OK with providing a more restrictive API to start and opening it up with more features. This makes it easier to provide a public API without throwing open the doors to all dispatch keys.

Reply:
The idea here is that, all keys except Autograd will still have the python dispatch function cached as an attribute, thus not supporting the dynamic behavior.